tests/test_session.py

```diff
@@ -295,14 +295,17 @@ def penncnp_cleanup(session):
 
 def test_session_browser_penncnp(slog, config_file, session, config_test_data, penncnp_cleanup):
     project = 'browser_penncnp'
-    # import pdb; pdb.set_trace()
-    server = session.connect_server(project, True)
-
-    assert server != None, "Error: could not connect server! Make sure " + project + " is correctly defined in " + config_file
+    with session.connect_server(project, True) as server:
+        assert server != None, "Error: could not connect server! Make sure " + project + " is correctly defined in " + config_file
 
-    wait = session.initialize_penncnp_wait()
-    assert session.get_penncnp_export_report(wait)
+        wait = session.initialize_penncnp_wait()
+        assert session.get_penncnp_export_report(wait), "Error: could not export report."
 
+def test_stale_penncnp(slog, config_file, session, config_test_data, penncnp_cleanup):
+    project = 'browser_penncnp'
+    with session.connect_server(project, True) as server:
+        with session.connect_server(project, True) as server2:
+            assert server2 != None, "Error: could not create second server"
```

Comment:
@gcyoung1 A couple questions:

1. would an `assert server == None` after line 308 or some other check to verify that `server` was terminated?  I'm assuming that when entering the context of `server2`, `server` would get terminated.  If yes, could you modify to validate that `server` has been killed? 
2. In a similar vein, should we verify that after exiting the `server2` context that selenium has exited?

Reply:
@jimklo 
1. Server is not terminated upon entering server2's context. Currently this test just fails, since the second attempt at creating a server detects the already running selenium process. We would need to remove the test for existing selenium processes to allow the second one to spawn. I'm not sure if that's something we even want to allow. So maybe I should just remove this test.
2. I think it would be better to make a separate test. Just create a single session then verify that the selenium process has been killed after we exit its context.